### DISCUSSION TITLE
Technical Workshop April 13 2024

### DISCUSSION THREAD

**Discussion by cmutel** [Depart-de-Sentier/sentier.dev#1] [Announcements]
# Sentier.dev First Technical Workshop 

## Logistics

* What: Discussion and development one-day workshop
* Where: Offices of eaternity.org: Siewerdtstrasse 95, 8050 Zürich, Switzerland
* When: 9:15 to 17:15 (plus drinks afterwards for those who want)

## Example data:

* Model: [US EPA WARM](https://www.epa.gov/warm/versions-waste-reduction-model#v16) (Be sure to show hidden sheets)
* Taxonomy - Products: [Combined Nomenclature 2024](https://showvoc.op.europa.eu/#/datasets/ESTAT_Combined_Nomenclature,_2024_%28CN_2024%29/data)
* Taxonomy - Processes: [NACE](https://showvoc.op.europa.eu/#/datasets/ESTAT_Statistical_Classification_of_Economic_Activities_in_the_European_Community_Rev._2.1._%28NACE_2.1%29/data)

Product overview:

![DdS concept overview](https://github.com/Depart-de-Sentier/sentier.dev/assets/645217/daf2b8a7-1164-49fd-a9b2-5b33f4cc2927)

Glossary concept:

<img width="851" alt="Screenshot 2024-04-13 at 09 24 15" src="https://github.com/Depart-de-Sentier/sentier.dev/assets/645217/6d738f7b-a128-4a4d-90d5-94b420b623ce">

### After lunch specific topic discussions

* Transport in the network: Implicit, includes transport means, cooling, losses, taxes and tariffs
* Result object validation, schema, intended purposes, needs
* Implementation of queues for calling models
* What about generic models (aside from transport), which change the state of a product (like freezing) but which don't have specific inputs or output but can operate on many difference kinds of products?

### Glossary data model, storage, changes

* [Glossary data storage](https://github.com/Depart-de-Sentier/sentier.dev/discussions/2)
* [Glossary revision workflow](https://github.com/Depart-de-Sentier/sentier.dev/discussions/3)

### Data storage

* [Do we store raw data or only normalized data](https://github.com/Depart-de-Sentier/sentier.dev/discussions/4)
* [Data storage API - what and how](https://github.com/Depart-de-Sentier/sentier.dev/discussions/5)
* [What are the technical requirements for a data warehouse](https://github.com/Depart-de-Sentier/sentier.dev/discussions/6)

### Data validation

* [What do the data structures look like? How are they stored and versioned?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/7)

### Model communication

* [What is the data transfer object for information flow between models?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/8)
* [DataFrames - what implementations are out there, do we want to use DataFrames at all?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/9)
* [Do we push for a standard set of base classes or go duck-typing from the start?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/13)

### Model operation

* [How does orchestration work?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/12)
* How do we construct an execution graph when our model dependency graph has circular references? (Not an issue, we can [visit nodes more than once](https://github.com/brightway-lca/bw_graph_tools/blob/d8ac6571f387c95459d72f15f8c81fbede3c5332/bw_graph_tools/graph_traversal.py#L159).
* [When do we drop down to the default, linear background matrix?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/14)
* [How we do we handle heterogeneous data inputs](https://github.com/Depart-de-Sentier/sentier.dev/discussions/15) (e.g. in our data source some inputs are correlated, and stored in a single DataFrame, while in other sources these inputs are modeled independently)?

### Organisation

* [Do we need to decide now on a design paradigm of federated versus centralized data and model storage?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/10)
* [Do we already start building workflows for using confidential data, or do we leave this for later?](https://github.com/Depart-de-Sentier/sentier.dev/discussions/11)

**Comment by mklarmann** [comment]
[Workshop.pdf](https://github.com/Depart-de-Sentier/sentier.dev/files/14967406/Workshop.pdf)
